src/main/kotlin/dev/akkinoc/spring/boot/logback/access/jetty/RequestWrapper.kt

```diff
@@ -0,0 +1,356 @@
+package dev.akkinoc.spring.boot.logback.access.jetty
+
+import ch.qos.logback.access.common.spi.IAccessEvent.NA
+import ch.qos.logback.access.common.spi.WrappedHttpRequest
+import jakarta.servlet.AsyncContext
+import jakarta.servlet.DispatcherType
+import jakarta.servlet.RequestDispatcher
+import jakarta.servlet.ServletConnection
+import jakarta.servlet.ServletContext
+import jakarta.servlet.ServletException
+import jakarta.servlet.ServletInputStream
+import jakarta.servlet.ServletRequest
+import jakarta.servlet.ServletResponse
+import jakarta.servlet.http.Cookie
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import jakarta.servlet.http.HttpSession
+import jakarta.servlet.http.HttpUpgradeHandler
+import jakarta.servlet.http.Part
+import org.eclipse.jetty.http.HttpHeader
+import org.eclipse.jetty.http.HttpScheme
+import org.eclipse.jetty.http.HttpURI
+import org.eclipse.jetty.http.HttpVersion
+import org.eclipse.jetty.server.Request
+import org.eclipse.jetty.server.Session
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.UnsupportedEncodingException
+import java.nio.charset.StandardCharsets
+import java.security.Principal
+import java.util.Collections
+import java.util.Enumeration
+import java.util.Locale
+import java.util.TreeMap
+
+/**
+ * Request wrapper that implements Servlet API.
+ */
+class RequestWrapper(private val request: Request) : HttpServletRequest, WrappedHttpRequest {

```

Comment:
It might be a better idea to ask the [logback people](https://github.com/qos-ch/logback-access) to make their `RequestWrapper` constuctor public.

Reply:
https://github.com/qos-ch/logback-access/issues/12